lld/MachO/Driver.cpp

```diff
@@ -1838,34 +1838,44 @@ bool link(ArrayRef<const char *> argsArr, llvm::raw_ostream &stdoutOS,
       if (const Arg *arg = args.getLastArgNoClaim(OPT_call_graph_profile_sort))
         error(firstArgStr + " is incompatible with " + arg->getSpelling());
   };
-  if (const Arg *arg = args.getLastArg(OPT_irpgo_profile_sort)) {
-    config->irpgoProfileSortProfilePath = arg->getValue();
+  if (args.hasArg(OPT_irpgo_profile_sort) ||
+      args.hasArg(OPT_irpgo_profile_sort_eq))
+    error("--irpgo-profile-sort is deprecated. Please use "
+          "--bp-startup-sort=function");
```

Comment:
Let's make this a warning so the flag is still supported. Also, these flag still need to set `config->irpgoProfilePath` and `config->bpStartupFunctionSort` so they still work.
```suggestion
    warn("--irpgo-profile-sort is deprecated. Please use "
          "--bp-startup-sort=function");
```

Reply:
@ellishg Because the lld testcase add the `-fatal_warnings` so if I change to the `log()`. Is it OK?

Reply:
You can use `%no-fatal-warnings-lld` instead `%lld` for the test case if you want to use lld without `-fatal_warnings`.